trusted_application_demo/ca_security_api/src/teec_utils.c

```diff
@@ -0,0 +1,136 @@
+
+void
+ca_teec_check_result(TEEC_Result res, const char * function, const char * errmsg)
+{
+  printf("\nRet: 0x%x ", res);
+  if (TEEC_SUCCESS != res) {
+    printf("[ERROR] in %s \n[MSG] %s: ", function, errmsg);
+    switch ((uint32_t)res) {
+      case TEEC_ERROR_GENERIC:
+        printf("TEEC_ERROR_GENERIC\n");
+      case TEEC_ERROR_ACCESS_DENIED:
+        printf("TEEC_ERROR_ACCESS_DENIED\n");
+        break;
+      case TEEC_ERROR_CANCEL:
+        printf("TEEC_ERROR_CANCEL\n");
+        break;
+      case TEEC_ERROR_ACCESS_CONFLICT:
+        printf("TEEC_ERROR_ACCESS_CONFLICT\n");
+        break;
+      case TEEC_ERROR_EXCESS_DATA:
+        printf("TEEC_ERROR_EXCESS_DATA\n");
+        break;
+      case TEEC_ERROR_BAD_FORMAT:
+        printf("TEEC_ERROR_BAD_FORMAT\n");
+        break;
+      case TEEC_ERROR_BAD_PARAMETERS:
+        printf("TEEC_ERROR_BAD_PARAMETERS\n");
+        break;
+      case TEEC_ERROR_BAD_STATE:
+        printf("TEEC_ERROR_BAD_STATE\n");
+        break;
+      case TEEC_ERROR_ITEM_NOT_FOUND:
+        printf("TEEC_ERROR_ITEM_NOT_FOUND\n");
+        break;
+      case TEEC_ERROR_NOT_IMPLEMENTED:
+        printf("TEEC_ERROR_NOT_IMPLEMENTED\n");
+        break;
+      case TEEC_ERROR_NOT_SUPPORTED:
+        printf("TEEC_ERROR_NOT_SUPPORTED\n");
+        break;
+      case TEEC_ERROR_NO_DATA:
+        printf("TEEC_ERROR_NO_DATA\n");
+        break;
+      case TEEC_ERROR_OUT_OF_MEMORY:
+        printf("TEEC_ERROR_OUT_OF_MEMORY\n");
+        break;
+      case TEEC_ERROR_BUSY:
+        printf("TEEC_ERROR_BUSY\n");
+        break;
+      case TEEC_ERROR_COMMUNICATION:
+        printf("TEEC_ERROR_COMMUNICATION\n");
+        break;
+      case TEEC_ERROR_SECURITY:
+        printf("TEEC_ERROR_SECURITY\n");
+        break;
+      case TEEC_ERROR_SHORT_BUFFER:
+        printf("TEEC_ERROR_SHORT_BUFFER\n");
+        break;
+      case TEEC_ERROR_EXTERNAL_CANCEL:
+        printf("TEEC_ERROR_EXTERNAL_CANCEL\n");
+        break;
+      case TEEC_ERROR_TARGET_DEAD:
+        printf("TEEC_ERROR_TARGET_DEAD\n");
+        break;
+      default:
+        printf("Unknown opcode\n");
+        break;
+    }
+  }
+  printf("SUCCESS: %s\n", errmsg);
+}
+
+TEEC_Result
+ca_teec_allocate_shared_memory(
+  TEEC_Context * ctx,
+  TEEC_SharedMemory * shm,
+  size_t sz,
+  uint32_t flags)
+{
+  TEEC_Result res;
+  shm->flags = flags;
+  shm->buffer = NULL;
+  shm->size = sz;
+  res = TEEC_AllocateSharedMemory(ctx, shm);
+  ca_teec_check_result(res, __func__, "TEEC_AllocateSharedMemory");
+  return res;
+}
+
+TEEC_Result
+ca_teec_open_session(
+  TEEC_UUID uuid,
+  TEEC_Context * ctx,
+  TEEC_Session * sess)
+{
+  TEEC_Result res;
+  uint32_t err_origin;
+
+  res = TEEC_InitializeContext(NULL, ctx);
+  ca_teec_check_result(res, __func__, "TEEC_InitializeContext");
+  if (TEEC_SUCCESS != res) {
+    return res;
+  }
+
+  res = TEEC_OpenSession(ctx,
+      sess,
+      &uuid,
+      TEEC_LOGIN_PUBLIC,
+      NULL,
+      NULL,
+      &err_origin);
```

Comment:
Style: 4-space indent instead of 2 ? Wrapping after the parenthesis would allow you to use 2 space indent everywhere (same everywhere below)

Reply:
The linter was giving me this coding style, so I did not change it.

Reply:
Try to wrap directly after the open parenthesis in line 119 *before* the first argument. Then all arguments should only be indented one level / 2 space.